frontend/src/metabase/lib/urls/utils.ts

```diff
@@ -28,3 +30,24 @@ export function getEncodedUrlSearchParams(query: Record<string, unknown>) {
     })
     .join("&");
 }
+
+export function getSubpathSafeUrl(url?: string) {
+  if (url === undefined) {
+    return;
+  }
+
+  return api.basename + url.trim();
```

Comment:
why trim?

Reply:
I thought we might want to remove whitespace and root out the strings that contain only whitespace but their length is > 0.
https://github.com/metabase/metabase/pull/46684/files#diff-f4a539fd7d0b7bd0e5169e7d39e57aee515d981a4366629ef2c2c162b23f2261R43

Reply:
this is strange, I would not expect such urls to be generated. This is not user input. Even if it was, it should be handled by consumers I believe this is not user input specific

Reply:
I made URLs non-optional strings and removed `.trim()`.
You might want to review this again:
https://github.com/metabase/metabase/pull/46684/commits/b017152bd13ff88bcfeb60563a351467a6b41a50